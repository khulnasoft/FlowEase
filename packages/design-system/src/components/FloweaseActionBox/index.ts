import FloweaseActionBox from './ActionBox.vue';

export default FloweaseActionBox;
