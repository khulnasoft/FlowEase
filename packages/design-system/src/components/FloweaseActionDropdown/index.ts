import FloweaseActionDropdown from './ActionDropdown.vue';
export default FloweaseActionDropdown;
