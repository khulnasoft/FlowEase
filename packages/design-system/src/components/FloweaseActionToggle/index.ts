import FloweaseActionToggle from './ActionToggle.vue';

export default FloweaseActionToggle;
