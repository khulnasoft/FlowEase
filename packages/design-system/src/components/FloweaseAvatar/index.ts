import FloweaseAvatar from './Avatar.vue';

export default FloweaseAvatar;
