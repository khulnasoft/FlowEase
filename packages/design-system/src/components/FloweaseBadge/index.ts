import FloweaseBadge from './Badge.vue';

export default FloweaseBadge;
