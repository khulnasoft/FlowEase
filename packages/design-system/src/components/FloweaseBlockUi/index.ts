import FloweaseBlockUi from './BlockUi.vue';

export default FloweaseBlockUi;
