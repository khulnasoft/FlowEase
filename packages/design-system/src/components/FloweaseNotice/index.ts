import FloweaseNotice from './Notice.vue';

export default FloweaseNotice;
