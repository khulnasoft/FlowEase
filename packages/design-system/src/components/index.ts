export { default as FloweaseActionBox } from './FloweaseActionBox';
export { default as FloweaseActionDropdown } from './FloweaseActionDropdown';
export { default as FloweaseActionToggle } from './FloweaseActionToggle';
export { default as FloweaseAlert } from './FloweaseAlert';
export { default as FloweaseAvatar } from './FloweaseAvatar';
export { default as FloweaseBadge } from './FloweaseBadge';
export { default as FloweaseBlockUi } from './FloweaseBlockUi';
export { default as FloweaseButton } from './FloweaseButton';
export { default as FloweaseCallout } from './FloweaseCallout';
export { default as FloweaseCard } from './FloweaseCard';
export { default as FloweaseCheckbox } from './FloweaseCheckbox';
export { default as FloweaseCircleLoader } from './FloweaseCircleLoader';
export { default as FloweaseColorPicker } from './FloweaseColorPicker';
export { default as FloweaseDatatable } from './FloweaseDatatable';
export { default as FloweaseFormBox } from './FloweaseFormBox';
export { default as FloweaseFormInputs } from './FloweaseFormInputs';
export { default as FloweaseFormInput } from './FloweaseFormInput';
export { default as FloweaseHeading } from './FloweaseHeading';
export { default as FloweaseIcon } from './FloweaseIcon';
export { default as FloweaseIconButton } from './FloweaseIconButton';
export { default as FloweaseInfoAccordion } from './FloweaseInfoAccordion';
export { default as FloweaseInfoTip } from './FloweaseInfoTip';
export { default as FloweaseInput } from './FloweaseInput';
export { default as FloweaseInputLabel } from './FloweaseInputLabel';
export { default as FloweaseInputNumber } from './FloweaseInputNumber';
export { default as FloweaseLink } from './FloweaseLink';
export { default as FloweaseLoading } from './FloweaseLoading';
export { default as FloweaseMarkdown } from './FloweaseMarkdown';
export { default as FloweaseMenu } from './FloweaseMenu';
export { default as FloweaseMenuItem } from './FloweaseMenuItem';
export { default as FloweaseNodeCreatorNode } from './FloweaseNodeCreatorNode';
export { default as FloweaseNodeIcon } from './FloweaseNodeIcon';
export { default as FloweaseNotice } from './FloweaseNotice';
export { default as FloweaseOption } from './FloweaseOption';
export { default as FloweasePopover } from './FloweasePopover';
export { default as FloweasePulse } from './FloweasePulse';
export { default as FloweaseRadioButtons } from './FloweaseRadioButtons';
export { default as FloweaseRecycleScroller } from './FloweaseRecycleScroller';
export { default as FloweaseResizeWrapper } from './FloweaseResizeWrapper';
export { default as FloweaseSelect } from './FloweaseSelect';
export { default as FloweaseSpinner } from './FloweaseSpinner';
export { default as FloweaseSticky } from './FloweaseSticky';
export { default as FloweaseTabs } from './FloweaseTabs';
export { default as FloweaseTag } from './FloweaseTag';
export { default as FloweaseTags } from './FloweaseTags';
export { default as FloweaseText } from './FloweaseText';
export { default as FloweaseTooltip } from './FloweaseTooltip';
export { default as FloweaseTree } from './FloweaseTree';
export { default as FloweaseUserStack } from './FloweaseUserStack';
export { default as FloweaseUserInfo } from './FloweaseUserInfo';
export { default as FloweaseUserSelect } from './FloweaseUserSelect';
export { default as FloweaseUsersList } from './FloweaseUsersList';
export { default as FloweaseResizeObserver } from './ResizeObserver';
export { FloweaseKeyboardShortcut } from './FloweaseKeyboardShortcut';
