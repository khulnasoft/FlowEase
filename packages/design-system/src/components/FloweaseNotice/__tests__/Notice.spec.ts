import { render } from '@testing-library/vue';
import FloweaseNotice from '../Notice.vue';
import { FloweaseText } from '@/components';

describe('components', () => {
	describe('FloweaseNotice', () => {
		it('should render correctly', () => {
			const wrapper = render(FloweaseNotice, {
				props: {
					id: 'notice',
				},
				slots: {
					default: 'This is a notice.',
				},
				global: {
					stubs: ['flowease-text'],
				},
			});
			expect(wrapper.html()).toMatchSnapshot();
		});

		describe('props', () => {
			describe('content', () => {
				it('should render correctly with content prop', () => {
					const wrapper = render(FloweaseNotice, {
						props: {
							id: 'notice',
							content: 'This is a notice.',
						},
						global: {
							stubs: ['flowease-text'],
						},
					});
					expect(wrapper.html()).toMatchSnapshot();
				});

				it('should render HTML', () => {
					const wrapper = render(FloweaseNotice, {
						props: {
							id: 'notice',
							content: '<strong>Hello world!</strong> This is a notice.',
						},
						global: {
							components: {
								'flowease-text': FloweaseText,
							},
						},
					});

					expect(wrapper.container.querySelectorAll('strong')).toHaveLength(1);
					expect(wrapper.html()).toMatchSnapshot();
				});

				it('should sanitize rendered HTML', () => {
					const wrapper = render(FloweaseNotice, {
						props: {
							id: 'notice',
							content: '<script>alert(1);</script> This is a notice.',
						},
						global: {
							stubs: ['flowease-text'],
						},
					});

					expect(wrapper.container.querySelector('script')).not.toBeTruthy();
					expect(wrapper.html()).toMatchSnapshot();
				});
			});
		});
	});
});
