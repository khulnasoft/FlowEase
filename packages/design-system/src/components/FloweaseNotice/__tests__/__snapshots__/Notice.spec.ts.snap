// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`components > FloweaseNotice > props > content > should render HTML 1`] = `
"<div id="notice" class="notice notice warning" role="alert">
  <div class="notice-content"><span class="flowease-text compact size-small regular"><span id="notice-content" class="truncated" role="region"><strong>Hello world!</strong> This is a notice.</span></span></div>
</div>"
`;

exports[`components > FloweaseNotice > props > content > should render correctly with content prop 1`] = `
"<div id="notice" class="notice notice warning" role="alert">
  <div class="notice-content">
    <flowease-text-stub bold="false" size="small" compact="true" tag="span"></flowease-text-stub>
  </div>
</div>"
`;

exports[`components > FloweaseNotice > props > content > should sanitize rendered HTML 1`] = `
"<div id="notice" class="notice notice warning" role="alert">
  <div class="notice-content">
    <flowease-text-stub bold="false" size="small" compact="true" tag="span"></flowease-text-stub>
  </div>
</div>"
`;

exports[`components > FloweaseNotice > should render correctly 1`] = `
"<div id="notice" class="notice notice warning" role="alert">
  <div class="notice-content">
    <flowease-text-stub bold="false" size="small" compact="true" tag="span"></flowease-text-stub>
  </div>
</div>"
`;
